feature-libs/organization/order-approval/components/details/order-approval-detail.service.ts

```diff
@@ -61,4 +61,19 @@ export class OrderApprovalDetailService {
   getOrderApproval(): Observable<OrderApproval> {
     return this.orderApproval$;
   }
+
+  /**
+   * Returns the approval details state
+   */
+  getOrderDetailsState(): Observable<StateUtils.LoaderState<Order>> {
+    return this.orderApproval$.pipe(
+      switchMap(() => this.getOrderDetails()),
+      map((order) => {
+        return {
+          error: false,
+          value: order,
+        } as StateUtils.LoaderState<Order>;
+      })
```

Comment:
why in this case it's hardcoded?

Reply:
it is because order approval has no state in store, but it is using the same approach as other services: `providers: [
            {
              provide: OrderDetailsService,
              useExisting: OrderApprovalDetailService,
            },
          ],`

so method should return `Observable<StateUtils.LoaderState<Order>>`

Reply:
Are you sure there is no state for order approvals?

`OrderApprovalService` looks like uses store